website/src/pages/StatusPage/StatusPage.tsx

```diff
@@ -32,23 +44,95 @@ export default function StatusPage() {
     `${import.meta.env.VITE_API_URL}recent`
   );
 
+  const [filters, setFilters] = useState({
+    type: '',
+    status: '',
+    source: '',
+  });
+
+  const filterData = (data: dataTypes[]) => {
+    return data.filter((item) => {
+      const itemType = item.type_of ? item.type_of.toString() : '';
+      const itemSource = item.source ? item.source.toString() : '';
+      const itemStatus = item.status ? item.status.toString() : '';
+
+      const matchesType =
+        filters.type === '' || filters.type === undefined || itemType === filters.type;
+      const matchesSource =
+        filters.source === '' || filters.source === undefined || itemSource === filters.source;
+      const matchesStatus =
+        filters.status === '' || filters.status === undefined || itemStatus === filters.status;
+
+      return (matchesType || filters.type === '' || filters.type === undefined) &&
+      (matchesSource || filters.source === '' || filters.source === undefined) &&
+      (matchesStatus || filters.status === '' || filters.status === undefined);
```

Comment:
This can be simplified. You are already checking if the variables are empty strings or undefined. The code below works:
```suggestion
      return matchesType && matchesSource && matchesStatus;
```

Reply:
resolved in 811693c